dttools/src/sh_popen.c

```diff
@@ -0,0 +1,129 @@
+#include "sh_popen.h"
+#include <time.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <sys/types.h>
+#include <errno.h>
+#include "stringtools.h"
+#include "itable.h"
+#include "debug.h"
+#include "xxmalloc.h"
+
+static struct itable* process_table = 0;
+
+FILE* sh_popen(char* command)
+{
+	pid_t pid;
+	int argc;
+	char** argv;
+	int fds[2];
+	char* cmd;
+	int result;
+
+	cmd = xxstrdup(command);
+
+	if(string_split_quotes(cmd, &argc, &argv) < 1){
```

Comment:
Why are you calling these function when you overwrite argc and argv below?

Reply:
it was an easier way to check and ensure that something resembling a command line command is being passed.

Reply:
If so, you need to free argv before the malloc below.